certbot-dns-cloudflare/certbot_dns_cloudflare/__init__.py

```diff
@@ -1 +1,76 @@
-"""Cloudflare DNS Authenticator"""
+"""
+The `~certbot_dns_cloudflare.dns_cloudflare` plugin automates the process of
+completing a ``dns-01`` challenge (`~acme.challenges.DNS01`) using the
+Cloudflare API.
+
+
+Named Arguments
+---------------
+
+========================================  =====================================
+``--dns-cloudflare-credentials``          Cloudflare credentials_ INI file.
+                                          (Required)
+``--dns-cloudflare-propagation-seconds``  The number of seconds to wait for DNS
+                                          to propagate before asking the ACME
+                                          server to verify the DNS record.
+                                          (Default: 10)
+========================================  =====================================
+
+Credentials
+-----------
+
+Use of this plugin requires a configuration file containing Cloudflare API
+credentials, obtained from your Cloudflare
+`account page <https://www.cloudflare.com/a/account/my-account>`_.
+
+.. code-block:: ini
+   :name: credentials.ini
+   :caption: Example credentials file:
+
+   # Cloudflare API credentials used by Certbot
+   dns_cloudflare_email = cloudflare@example.com
+   dns_cloudflare_api_key = 0123456789abcdef0123456789abcdef01234567
+
+The path to this file can be provided interactively or using the
+``--dns-cloudflare-credentials`` command-line argument. Certbot records the path
+to this file for use during renewal, but does not store the file's contents.
+
+.. caution::
+   You should protect these API credentials as you would the password to your
+   Cloudflare account. Users who can read this file can use these credentials
+   to issue API calls on your behalf. Users who can cause Certbot to run using
+   these credentials can complete a ``dns-01`` challenge to acquire new
+   certificates or revoke existing certificates for associated domains.
+
```

Comment:
I agree a dedicated, locked down machine is probably the best option, but I'm not sure if that's what we should recommend to the average user (although perhaps we could mention in an offhand comment that's a way to provide additional security here).

For these files in general, I think we should provide instructions that the user should follow largely regardless of the machine they actually end up running Certbot on it. For example:

1. Minimal permissions on file containing API credentials (e.g. 600).
2. Minimal permissions with API credentials itself. In the rare cases it is possible, we should instruct the user to create the API credentials with the fewest permissions Certbot needs to operate.

I believe we decided when we had this conversation before we weren't going to password protect or encrypt the file as it doesn't provide much additional security. Is there anything else we should do?

To cc people who were interested in this issue before:

cc @joohoi, @jsha